app/src/main/java/com/keylesspalace/tusky/adapter/AccountViewHolder.kt

```diff
@@ -0,0 +1,60 @@
+package com.keylesspalace.tusky.adapter
+
+import android.view.View
+import androidx.recyclerview.widget.RecyclerView
+import com.keylesspalace.tusky.R
+import com.keylesspalace.tusky.databinding.ItemAccountBinding
+import com.keylesspalace.tusky.entity.TimelineAccount
+import com.keylesspalace.tusky.interfaces.AccountActionListener
+import com.keylesspalace.tusky.interfaces.LinkListener
+import com.keylesspalace.tusky.util.emojify
+import com.keylesspalace.tusky.util.loadAvatar
+
+class AccountViewHolder(
+    private val binding: ItemAccountBinding
+) : RecyclerView.ViewHolder(binding.root) {
+    private var accountId: String? = null
+
+    fun setupWithAccount(
+        account: TimelineAccount,
+        animateAvatar: Boolean,
+        animateEmojis: Boolean,
+        showBotOverlay: Boolean
+    ) {
+        accountId = account.id
+
+        val format = binding.accountUsername.context.getString(R.string.post_username_format)
+        val formattedUsername = String.format(format, account.username)
+        binding.accountUsername.text = formattedUsername
+
+        val emojifiedName = account.name.emojify(
+            account.emojis,
+            binding.accountDisplayName,
+            animateEmojis
+        )
+        binding.accountDisplayName.text = emojifiedName
+
+        val avatarRadius = binding.accountAvatar.context.resources
+            .getDimensionPixelSize(R.dimen.avatar_radius_48dp)
+        loadAvatar(account.avatar, binding.accountAvatar, avatarRadius, animateAvatar)
+
+        if (showBotOverlay && account.bot) {
+            binding.accountAvatarInset.visibility = View.VISIBLE
+            binding.accountAvatarInset.setImageResource(R.drawable.bot_badge)
```

Comment:
could set that in xml already and then just do
`binding.accountAvatarInset.visible(showBotOverlay && account.bot)`

Maybe even call `accountAvatarInset` `accountBotBadge`.

Reply:
Done.